### DISCUSSION TITLE
Docking windows?

### DISCUSSION THREAD

**Discussion by Sahasrara** [tauri-apps/tauri#8250] [Q&A] [answered]
Is there any way to create multiple windows with Tauri and then have them be "dockable" with one another?

Here's an example of what I mean:
[Dear IMGUI Docking Example](https://user-images.githubusercontent.com/8225057/46304087-00035580-c5ae-11e8-8904-f27a9434574a.gif)

If not, and this is probably a crazy question, how difficult would it be to swap the backend from using tao/GTK to using GLFW so that I can use Dear IMGUI for docking Tauri "windows".

**Comment by FabianLars** [comment, marked as answer]
> Is there any way to create multiple windows with Tauri and then have them be "dockable" with one another?

Right now i'd say it's not reallyyy possible. I think it can be implemented sometime in v2 though which will support multiple webviews at some point (implemented in Wry, not used in Tauri yet) and better drag and drop apis (still being worked on).

> If not, and this is probably a crazy question, how difficult would it be to swap the backend from using tao/GTK to using GLFW so that I can use Dear IMGUI for docking Tauri "windows".

Sounds impossible tbh, even now where wry webviews aren't really coupled to the window libraries anymore.


Anyway, i agree that this would be a cool feature. We probably can't ever get close to how insane imgui does it with its custom renderer but _some_ form of docking for sure will be possible at some point :)

**Reply by Sahasrara** [reply]
Three follow-up questions:

1. Where can I follow the progress of V2 multiple webviews and drag and drop?
2. If I wanted to contribute, how would I do that?
3. I had one more thought on how this might be achieved, but it's kind of hacky. I wanted to get your opinion:

I could render dear imgui windows using the rust bindings. One window for each corresponding Tauri window. Then, from the dear imgui rendering code, I could force position the Tauri windows to perfectly match the location of the dear imgui windows. Does that seem crazy?

**Reply by FabianLars** [reply]
> Where can I follow the progress of V2 multiple webviews

Hard to say right now since the work (in tauri) didn't really start yet afaik.

> drag and drop?

Since it's client work it's implemented a bit differently and in private first, but i'm sure https://github.com/tauri-apps/tauri/issues/2593 will get a few updates.

> If I wanted to contribute, how would I do that?

In what context? For one of those features? They are quite difficult topics so i'd recommend playing around with tao/wry/tauri codebases first.
For drag-drop we'll have to wait until it's public etc, but iirc the planned work makes it mostly feature complete.

> I had one more thought on how this might be achieved, but it's kind of hacky. I wanted to get your opinion:

"Does that seem crazy?" - Honestly? Yes :D I'm not sure if i 100% get what you have in mind. I don't see how docking/undocking would work there.

**Reply by Sahasrara** [reply]
Right-o, thanks very much for all your help! That answers all my questions.

**Reply by alexgeek** [reply]
Sorry to gravedig but since it's been a while, have their been any developments?

**Reply by spencer741** [reply]
Bump. Would also be interested in any developments / where things are lacking today.  @FabianLars

**Reply by FabianLars** [reply]
initial multiwebview support landed since then, guarded behind the `unstable` feature flag. I wouldn't really recommend using it much yet, especially not if you can't handle not having any docs.
